compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4508,6 +4514,26 @@ private void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr) {
             exprExpType = BUnionType.create(null, expType, symTable.errorType);
         }
 
+        if (checkedExpr.expr.getKind() == NodeKind.FIELD_BASED_ACCESS_EXPR &&
+                checkedExpr.getKind() == NodeKind.CHECK_EXPR && types.isUnionOfSimpleBasicTypes(expType)) {
+            ArrayList<BLangExpression> argExprs = new ArrayList<>();
```

Comment:
In this method, `checkedExpr.expr` and `checkedExpr.getKind()` used several times, Shall we extract both as separate variables.

Reply:
Done.